integration-tests/elasticsearch-rest-client/src/test/java/io/quarkus/it/elasticsearch/FruitResourceTest.java

```diff
@@ -17,41 +21,62 @@ public class FruitResourceTest {
     };
 
     @Test
-    public void testEndpoint() throws InterruptedException {
+    public void testEndpoint() {
         // create a Fruit
         Fruit fruit = new Fruit();
         fruit.id = "1";
         fruit.name = "Apple";
         fruit.color = "Green";
-        given()
-                .contentType("application/json")
-                .body(fruit)
-                .when().post("/fruits")
-                .then()
-                .statusCode(201);
+
+        await().atMost(1, TimeUnit.SECONDS).untilAsserted(() -> {
+            given()
+                    .contentType("application/json")
+                    .body(fruit)
+                    .when().post("/fruits")
+                    .then()
+                    .statusCode(201);
+        });
 
         // get the Fruit
         Fruit result = get("/fruits/1").as(Fruit.class);
-        Assertions.assertNotNull(result);
-        Assertions.assertEquals("1", result.id);
-        Assertions.assertEquals("Apple", result.name);
-        Assertions.assertEquals("Green", result.color);
 
-        // wait a few ms for the indexing to happened
-        Thread.sleep(1000);
+        assertThat(result).isNotNull().isEqualTo(fruit);
 
         // search the Fruit
         List<Fruit> results = get("/fruits/search?color=Green").as(LIST_OF_FRUIT_TYPE_REF);
-        Assertions.assertNotNull(results);
-        Assertions.assertFalse(results.isEmpty());
-        Assertions.assertEquals("1", results.get(0).id);
-        Assertions.assertEquals("Apple", results.get(0).name);
-        Assertions.assertEquals("Green", results.get(0).color);
+        assertThat(results).hasSize(1).contains(fruit);
+
         results = get("/fruits/search?name=Apple").as(LIST_OF_FRUIT_TYPE_REF);
-        Assertions.assertNotNull(results);
-        Assertions.assertFalse(results.isEmpty());
-        Assertions.assertEquals("1", results.get(0).id);
-        Assertions.assertEquals("Apple", results.get(0).name);
-        Assertions.assertEquals("Green", results.get(0).color);
+        assertThat(results).hasSize(1).contains(fruit);
+
+        //create new fruit index via bulk operation
+        Fruit pomegranate = new Fruit();
+        pomegranate.id = "2";
+        pomegranate.name = "Pomegranate";
+        pomegranate.color = "Red";
+
+        List<Fruit> fruits = List.of(pomegranate);
+
+        await().atMost(1, TimeUnit.SECONDS).untilAsserted(() -> {
+            given()
+                    .contentType("application/json")
+                    .body(fruits)
+                    .when().post("/fruits/bulk")
+                    .then()
+                    .statusCode(200);
+        });
+        result = get("/fruits/2").as(Fruit.class);
+        assertThat(result).isNotNull().isEqualTo(pomegranate);
```

Comment:
```suggestion
        given()
                .contentType("application/json")
                .body(fruits)
                .when().post("/fruits/bulk")
                .then()
                .statusCode(200);
                
        await().atMost(1, TimeUnit.SECONDS).untilAsserted(() -> {
                result = get("/fruits/2").as(Fruit.class);
                assertThat(result).isNotNull().isEqualTo(pomegranate);
        });
```
the bulk operation will return 200 right away (if all goes well in general 😉) it's the get operation after the bulk that may have a slight delay before it starts returning the indexed fruit.

As for the import/formatting issue that the builds reports.. try 
```
mvn impsort:sort formatter:format -pl :quarkus-integration-test-elasticsearch-java-client,:quarkus-integration-test-elasticsearch-rest-client
```

and then instead of merging the main branch you can do a rebase and squash all those commits to address formatting and other issues 😃.

Reply:
Thank you for your advice. Honestly, I'm just confused.
This happened when merge conflicts appeared. In vain I created a new branch and merged main into it. Lack of experience(